--- CMakeLists.txt.orig	2017-11-13 22:37:28.115678505 +0800
+++ CMakeLists.txt	2017-11-13 22:37:39.715678579 +0800
@@ -66,7 +66,7 @@
     add_definitions("-DXMR_STAK_THREADS=${XMR-STAK_THREADS}")
 endif()
 
-set(CUDA_ARCH "20;30;35;37;50;52;60;61;62" CACHE STRING "Set GPU architecture (semicolon separated list, e.g. '-DCUDA_ARCH=20;35;60')")
+set(CUDA_ARCH "30;35;37;50;52;60;61;62" CACHE STRING "Set GPU architecture (semicolon separated list, e.g. '-DCUDA_ARCH=35;60')")
 
 # validate architectures (only numbers are allowed)
 foreach(CUDA_ARCH_ELEM ${CUDA_ARCH})
